lighthouse-cli/index.js

```diff
@@ -139,22 +142,47 @@ if (cli.verbose) {
   flags.logLevel = 'error';
 }
 
+function run(retries) {
```

Comment:
I like the setup here to default to trying to see if a Chrome we can attach to is already running, but the flow ends up somewhat hard to follow, especially the flow of errors through the conditionals, two `catch` blocks and the recursive `run()` call :)

Can you separate this a bit (into separate functions?) to make the flow clearer? This is only going to get more dire as #640 *also* wants to run this recursively and #639 reworks how errors bubble up to the CLI